### ISSUE TITLE
Unable use different EdmModels for different routes

### ISSUE THREAD

**Issue by wierzba3** [OData/WebApi#1533]
We currently have one shared EdmModel for all of our route prefixes, but we desire two. 

For one route prefix, we wish to explicitly block ( return 404) for a subset of the API functionality described in the EdmModel. We wish to define a second EdmModel that only describes this subset of functionality. It seems that this is not supported, because we are receiving errors when we do this.

Full, detailed description with minified example:
[Stackoverflow question](https://stackoverflow.com/questions/51314002/expose-only-a-subset-of-net-odata-apis-to-a-route-using-an-additional-edmmodel)

Library version:
Microsoft.AspNet.OData.5.9.0

### Reproduce steps
See stackoverflow question: [Stackoverflow question](https://stackoverflow.com/questions/51314002/expose-only-a-subset-of-net-odata-apis-to-a-route-using-an-additional-edmmodel)

Basically, to repro: Create two EdmModels in one service. One EdmModel should only expose a subset of the APIs compared to the other. Map one model to one route prefix, map the other EdmModel to the other route prefix. On service startup, you will see an error, something like:

The path template 'myEntity' on the action 'MyAction' in controller 'MyEntityController' is not a valid OData path template. Resource not found for the segment 'myEntity'.)

### Expected result
I would expect the service to start without an error, and for the service to return 404, when trying to access one of these APIs that do not exist. Perhaps this is a misunderstanding on my part of how EdmModel mapping works.

### Actual result

Service crashes on startup due to error.

### Additional detail

n/a

**Comment by wierzba3** [comment]
Closing this issue and opening another one with a clearer description
